subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/execution/PluginsBlockInterpreter.kt

```diff
@@ -71,46 +75,66 @@ fun interpret(program: Program.Plugins): PluginsBlockInterpretation = KotlinLexe
 
             BLOCK_COMMENT -> Unit
             DOC_COMMENT -> Unit
+            EOL_COMMENT -> {
+                seenStatementSeparator = true
+            }
 
-            in WHITE_SPACE_OR_COMMENT_BIT_SET -> {
-                // detect newline separators between statements
-                if (state == InterpreterState.AFTER_ID && tokenSequence.contains('\n')) {
-                    newStatement()
+            in WHITESPACES -> {
+                if (tokenSequence.contains('\n')) {
+                    seenStatementSeparator = true
                 }
             }
 
             else -> {
                 when (state) {
+
                     InterpreterState.START -> state = when (tokenType) {
                         LBRACE -> InterpreterState.ID
                         else -> return expecting("{")
                     }
 
                     InterpreterState.ID -> state = when (tokenType) {
                         RBRACE -> InterpreterState.END
-                        SEMICOLON -> InterpreterState.ID
-                        IDENTIFIER -> when (tokenText) {
-                            "id" -> InterpreterState.ID_OPEN_CALL
-                            "kotlin" -> InterpreterState.KOTLIN_ID_OPEN_CALL
-                            else -> return expecting("id or kotlin")
+                        SEMICOLON -> InterpreterState.ID.also { if (pluginId != null) newPluginRequest() }
+                        IDENTIFIER -> {
+                            if (pluginId != null && seenStatementSeparator) {
+                                newPluginRequest()
+                            }
+                            when (tokenText) {
+                                "id" -> InterpreterState.ID_OPEN_CALL
+                                "kotlin" -> InterpreterState.KOTLIN_ID_OPEN_CALL
+                                else -> return expecting("id or kotlin")
+                            }
                         }
 
                         else -> return expecting("plugin spec")
                     }
 
                     InterpreterState.AFTER_ID -> {
                         state = when (tokenType) {
-                            DOT -> state
-                            SEMICOLON -> newStatement()
-                            RBRACE -> {
-                                newStatement()
-                                InterpreterState.END
-                            }
-
-                            IDENTIFIER -> when (tokenText) {
-                                "version" -> InterpreterState.VERSION_OPEN_CALL
-                                "apply" -> InterpreterState.APPLY_OPEN_CALL
-                                else -> return expecting("version or apply")
+                            DOT -> InterpreterState.AFTER_ID
```

Comment:
🧪 This test case came to mind:

```kotlin
    @Test
    fun `syntax error - dot after dot`() {
        assertDynamicInterpretationOf(
            """id("plugin-id-1"). .version("1.0")""",
            "Expecting version or apply, got '.'"
        )
    }
```

Reply:
Good catch, it indeed didn't work. Fixed